backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/webservice/AcceptingStoresMutationService.kt

```diff
@@ -1,17 +1,76 @@
 import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import graphql.schema.DataFetchingEnvironment
+import org.jetbrains.exposed.sql.select
 import org.jetbrains.exposed.sql.transactions.transaction
+import org.slf4j.LoggerFactory
 
 @Suppress("unused")
 class AcceptingStoresMutationService {
     @GraphQLDescription("Import accepting stores via csv")
-    fun importAcceptingStores(stores: List<CSVAcceptingStore>): Boolean {
-        transaction {
-            // TODO #1571 store the store data in the database
-            print(stores)
+    fun importAcceptingStores(stores: List<CSVAcceptingStore>, project: String, dryRun: Boolean, dfe: DataFetchingEnvironment): StoreImportResultModel {
+        val logger = LoggerFactory.getLogger(Mailer::class.java)
+        val context = dfe.getContext<GraphQLContext>()
+        val jwtPayload = context.enforceSignedIn()
+        return transaction t@{
+            val projectEntity =
+                ProjectEntity.find { Projects.project eq project }.firstOrNull() ?: throw ProjectNotFoundException(
+                    project
+                )
+            val user = AdministratorEntity.findById(jwtPayload.adminId)
+                ?: throw UnauthorizedException()
+
+            if (!Authorizer.mayUpdateStoresInProject(user, projectEntity.id.value)) {
+                throw ForbiddenException()
+            }
+
+            try {
+                var numStoresCreated = 0
+                var numStoresUntouched = 0
+                val acceptingStoreIdsToRemove =
+                    AcceptingStores.slice(AcceptingStores.id).select { AcceptingStores.projectId eq projectEntity.id }
+                        .map { it[AcceptingStores.id].value }.toMutableSet()
+
+                for (acceptingStore in stores) {
+                    // If an exact duplicate is found in the DB, we do not recreate it and instead
+                    // remove the id from `acceptingStoreIdsToRemove`.
+                    val existingStoreId: Int? =
+                        AcceptingStoresRepository.determineRemovableAcceptingStoreId(mapCsvToStore(acceptingStore), projectEntity)
+                    if (existingStoreId != null) {
+                        acceptingStoreIdsToRemove.remove(existingStoreId)
+                        numStoresUntouched += 1
+                        continue
+                    }
+                    if (!dryRun) {
+                        AcceptingStoresRepository.createStore(mapCsvToStore(acceptingStore), projectEntity)
+                    }
+                    numStoresCreated += 1
+                }
+                if (!dryRun) {
+                    AcceptingStoresRepository.deleteStores(acceptingStoreIdsToRemove)
+                }
+                return@t StoreImportResultModel(numStoresCreated, acceptingStoreIdsToRemove.size, numStoresUntouched)
+            } catch (e: DatabaseIOException) {
```

Comment:
I guess this is an unreachable catch, this type of exception is thrown only inside this catch block. how can it be caught here?

Reply:
okay you are right. do i need here a try catch?
I think within a transaction if anything fails a rollback will happen?

Reply:
> do i need here a try catch?

I don't think so

> within a transaction if anything fails a rollback will happen?

yes